dotCMS/src/integration-test/java/com/dotcms/cache/lettuce/LettuceCacheTest.java

```diff
@@ -0,0 +1,153 @@
+package com.dotcms.cache.lettuce;
+
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.UUIDGenerator;
+
+public class LettuceCacheTest {
+    
+    
+    static RedisCache cache = null;
+    
+    @BeforeClass
+    public static void startup() throws Exception {
+        cache = new RedisCache();
+    }
+    
+    
+    /**
+     * all key values that are put into redis are prefixed by a random key.
+     * When a flushall is called, this key is cycled, which basically invalidates
+     * all cached entries.  The prefix key is stored in redis itself so multiple 
+     * servers in the c
+     */
+    @Test
+    public void test_prefix_key_cycling() {
+       final String prefix1  = cache.loadPrefix();
+       assert(prefix1!=null && prefix1.length()>3);
+       
+       cache.cycleKey();
+       
+       final String prefix2  = cache.loadPrefix();
+       assert(prefix2!=null);
+       
+       // keys don't match after they have been cycled
+       assert(!prefix1.equals(prefix2));
+       
+       // getting the key again and the keys match
+       final String prefix3  = cache.loadPrefix();
+       assert(prefix2.equals(prefix3));
+    }
+
+    @Test
+    public void test_cache_key_generation() {
+        
+        final String prefix  = cache.loadPrefix();
+        assert(prefix!=null && prefix.length()>3);
+        String group = "group";
+        String key = "key";
+        String cacheKey = cache.cacheKey(group, key);
+        assert (cacheKey!=null);
+        assert (cacheKey.startsWith(prefix));
```

Comment:
These asserts: `assert (cacheKey.startsWith(prefix));` and `assert (cacheKey.startsWith(prefix+ "." + group + "."));` are unnecessary if on the next line you write: `assert (cacheKey.equals(prefix+ "." + group+ "." + key));`